pkg/v1/cli/pluginmanager/manager.go

```diff
@@ -154,16 +168,60 @@ func AvailablePlugins(serverName string) ([]plugin.Discovered, error) {
 	setAvailablePluginsStatus(availablePlugins, installedSeverPluginDesc)
 	setAvailablePluginsStatus(availablePlugins, installedStandalonePluginDesc)
 
+	installedButNotDiscoveredPlugins := getInstalledButNotDiscoveredStandalonePlugins(availablePlugins, installedStandalonePluginDesc)
+	availablePlugins = append(availablePlugins, installedButNotDiscoveredPlugins...)
+
 	return availablePlugins, nil
 }
 
+func getInstalledButNotDiscoveredStandalonePlugins(availablePlugins []plugin.Discovered, installedPluginDesc []cliv1alpha1.PluginDescriptor) []plugin.Discovered {
+	var newPlugins []plugin.Discovered
+	for i := range installedPluginDesc {
+		found := false
+		for j := range availablePlugins {
+			if installedPluginDesc[i].Name == availablePlugins[j].Name {
+				found = true
+				// If plugin is installed but marked as not installed as part of availablePlugins list
+				// mark the plugin as installed
+				// This is possible if use has used --local mode to install the plugin which is also
+				// getting discovered from the configured discovery sources
+				if availablePlugins[j].Status == common.PluginStatusNotInstalled {
+					availablePlugins[j].Status = common.PluginStatusInstalled
+				}
+			}
+		}
+		if !found {
+			p := DiscoveredFromPluginDescriptor(&installedPluginDesc[i])
+			p.Scope = common.PluginScopeStandalone
+			p.Status = common.PluginStatusInstalled
+			newPlugins = append(newPlugins, p)
+		}
+	}
+	return newPlugins
+}
+
+// DiscoveredFromPluginDescriptor returns discovered plugin object from k8sV1alpha1
+func DiscoveredFromPluginDescriptor(p *cliv1alpha1.PluginDescriptor) plugin.Discovered {
+	dp := plugin.Discovered{
+		Name:               p.Name,
+		Description:        p.Description,
+		RecommendedVersion: p.Version,
```

Comment:
Should this include DiscoveryType, or is it even knowable here?

Reply:
No. We will not get the `DiscoveryType` information from the plugin descriptor.